src/custom/state/affiliate/hooks.ts

```diff
@@ -1,33 +1,13 @@
 import { useCallback } from 'react'
-import { useSelector } from 'react-redux'
-import { AppState } from 'state'
-import { useAppDispatch } from 'state/hooks'
-import { updateReferralAddress } from 'state/affiliate/actions'
-import { APP_DATA_HASH } from 'constants/index'
-
-export function useAppDataHash() {
-  return useSelector<AppState, string>((state) => {
-    return state.affiliate.appDataHash || APP_DATA_HASH
-  })
-}
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { setReferralAddressActive, updateReferralAddress } from 'state/affiliate/actions'
 
 export function useReferralAddress() {
-  return useSelector<
-    AppState,
-    | {
-        value: string
-        isValid: boolean
-      }
-    | undefined
-  >((state) => {
-    return state.affiliate.referralAddress
-  })
+  return useAppSelector((state) => state.affiliate.referralAddress)
 }
 
 export function useAddress() {
```

Comment:
not from this PR, but shouldn't the name be `useAffiliateAddress`? more coherent with `useReferralAddress` and also when imported this hook, will be a too general name in whichever component (so the reader won't related it to the affiliate program)

Reply:
Will address in a follow up PR

Reply:
https://github.com/cowprotocol/cowswap/pull/748